### PR TITLE
fix: specify source when document is from the url

### PR DESCRIPTION
**Pull request by magicmatatjahu**
**Description**

Specify source when document is from the url - now all refs are resolved from url point of view.

cc @derberg 

**Related issue(s)**
Fixes https://github.com/asyncapi/studio/issues/392

### CODE REVIEW

**Comment by derberg** [comment]
In UI it looks nice, clear info what file is rendered, and when I edit, localStorage is shown again. Thanks

side note, there is some other strange bug but you probably want to solve it separately. When I start editing, of course, local storage is used and all suddenly validation fails, because `base` changes. That is fine. But error about issue with dereferencing is not something I would expect: `TypeError: i.readFile is not a function`

**Comment by magicmatatjahu** [comment]
@derberg I changed message to the:

image

Do you like it?

**Comment by derberg** [comment]
Nice

small suggestion: if this is custom error for Studio only, then I suggest you write `File references are not yet supported in Studio` to make it clear that outside studio it works completely fine

File: src/services/specification.service.ts [hunk 1]
```diff
@@ -34,12 +34,24 @@ export class SpecificationService {
...
 
     try {
-      const { document, diagnostics: validationDiagnostics, extras } = await parser.parse(rawSpec);
-      diagnostics = validationDiagnostics;
+      const { document, diagnostics: validationDiagnostics, extras } = await parser.parse(rawSpec, options);
+
+      // map messages of invalid ref to file
+      diagnostics = validationDiagnostics.map(d => {
+        if (d.code === 'invalid-ref' && d.message.endsWith('readFile is not a function')) {
+          d.message = 'File references are not yet supported';
```
Comment (magicmatatjahu) [inline src/services/specification.service.ts:51]:
```suggestion
          d.message = 'File references are not yet supported in Studio';
```

**Comment by magicmatatjahu** [comment]
@derberg Done :)
@fmvilas Could you check?

**Comment by fmvilas** [comment]
Is this problem known?

Captura de Pantalla 2022-11-24 a las 17 00 00

**Comment by magicmatatjahu** [comment]
@fmvilas LOL, I haven't seen this error before. I tried replicating it locally and got nothing like that.... On the other hand, in the preview it occurs - presumably `monaco-yaml` diagnostic options can't download the schema from http... and `monaco-json` can.... I will create issue because it is not a serious problem, and autocompletion works, and it is the most important thing in json/yaml integration

Besides that, everything is ok in this PR?

**Comment by magicmatatjahu** [comment]
@fmvilas https://github.com/asyncapi/studio/issues/509

**Review by fmvilas — APPROVED** [review]
Yeah, all good with this PR. This error is already happening in master so 👍

**Pull request merged**